lib/model.js

```diff
@@ -4062,9 +4062,9 @@ Model.aggregate = function aggregate(pipeline, callback) {
  */
 
 Model.validate = function validate(obj, pathsToValidate, context, callback) {
-  if (context === void 0 && obj instanceof Model) {
-    // For convenience, if we're validating a document, make `context` default to
-    // the model so users don't have to always pass `context`, re: gh-10132.
+  if (context === void 0) {
```

Comment:
I think we should make this only happen if the user doesn't specify `context`. So either `arguments.length < 3`, or `arguments.length === 3 && typeof arguments[2] === 'function'`.